ursine/header.py

```diff
@@ -0,0 +1,106 @@
+import copy
+import random
+import string
+from .header_parsing import parse_header
+
+
+def random_tag():
+    return ''.join([random.choice(string.hexdigits) for _ in range(16)])
```

Comment:
You don't need list compehension, a generator will work.

Reply:
You also have `random.choices` now (as of 3.6): https://docs.python.org/3/library/random.html

```python
return ''.join(random.choices(string.hexdigits, k=16))
```